File: src/test/java/no/nav/familie/integrasjoner/infotrygd/InfotrygdTestConfig.java
package no.nav.familie.integrasjoner.infotrygd;

import no.nav.familie.integrasjoner.client.rest.InfotrygdClient;
import no.nav.familie.integrasjoner.infotrygd.domene.AktivKontantstøtteInfo;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;
import org.springframework.context.annotation.Profile;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.*;

@Configuration
public class InfotrygdTestConfig {

    @Bean
    @Profile("mock-infotrygd")
    @Primary
    public InfotrygdClient infotrygdServiceMock() {
        InfotrygdClient infotrygdClientMock = mock(InfotrygdClient.class);
        var aktivKontantstøtteInfo = new AktivKontantstøtteInfo(false);
        when(infotrygdClientMock.hentAktivKontantstøtteFor(any())).thenReturn(aktivKontantstøtteInfo);
        doNothing().when(infotrygdClientMock).ping();
        return infotrygdClientMock;
    }
}

File: src/test/java/no/nav/familie/integrasjoner/infotrygd/InfotrygdTestConfig.kt
package no.nav.familie.integrasjoner.infotrygd

import io.mockk.every
import io.mockk.mockk
import no.nav.familie.integrasjoner.client.rest.InfotrygdRestClient
import no.nav.familie.integrasjoner.infotrygd.domene.AktivKontantstøtteInfo
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile

@Configuration
class InfotrygdTestConfig {
    @Bean
    @Profile("mock-infotrygd")
    @Primary
    fun infotrygdRestClientMock(): InfotrygdRestClient {
        val infotrygdServiceMock: InfotrygdRestClient = mockk(relaxed = true)
        val aktivKontantstøtteInfo = AktivKontantstøtteInfo(false)
        every { infotrygdServiceMock.hentAktivKontantstøtteFor(any()) }
            .returns(aktivKontantstøtteInfo)
        return infotrygdServiceMock
    }
}
